plugins/sbt/src/main/scala/io/github/kobylynskyi/graphql/codegen/GraphQLCodegenSbtPlugin.scala

```diff
@@ -88,51 +74,72 @@ object GraphQLCodegenSbtPlugin extends AutoPlugin {
   // This plugin is automatically enabled for projects which are JvmPlugin.
   override def trigger = allRequirements
 
+  override val globalSettings = Seq(
+    graphqlModelNamePrefix := None,
+    graphqlModelNameSuffix := None,
+    graphqlApiPackageName := None,
+    graphqlModelPackageName := None,
+    graphqlGenerateBuilder := DefaultMappingConfigValues.DEFAULT_BUILDER,
+    graphqlGenerateApis := DefaultMappingConfigValues.DEFAULT_GENERATE_APIS,
+    graphqlModelValidationAnnotation := Some(DefaultMappingConfigValues.DEFAULT_VALIDATION_ANNOTATION),
+    graphqlGenerateEqualsAndHashCode := DefaultMappingConfigValues.DEFAULT_EQUALS_AND_HASHCODE,
+    graphqlGenerateToString := DefaultMappingConfigValues.DEFAULT_TO_STRING,
+    graphqlGenerateAsyncApi := DefaultMappingConfigValues.DEFAULT_GENERATE_ASYNC_APIS,
+    graphqlGenerateParameterizedFieldsResolvers := DefaultMappingConfigValues.DEFAULT_GENERATE_PARAMETERIZED_FIELDS_RESOLVERS,
+    graphqlGenerateExtensionFieldsResolvers := DefaultMappingConfigValues.DEFAULT_GENERATE_EXTENSION_FIELDS_RESOLVERS,
+    graphqlGenerateDataFetchingEnvArgInApis := DefaultMappingConfigValues.DEFAULT_GENERATE_DATA_FETCHING_ENV,
+    graphqlGenerateRequests := DefaultMappingConfigValues.DEFAULT_GENERATE_REQUESTS,
+    graphqlRequestSuffix := Some(DefaultMappingConfigValues.DEFAULT_REQUEST_SUFFIX),
+    graphqlResponseProjectionSuffix := Some(DefaultMappingConfigValues.DEFAULT_RESPONSE_PROJECTION_SUFFIX)
+  )
+
   // a group of settings that are automatically added to projects.
   override val projectSettings =
-    inConfig(Compile)(baseGraphQLSettings) ++
-    inConfig(Test)(baseGraphQLSettings)
+    inConfig(Compile)(baseGraphQLSettings)
+    // For now we skip compiling schemas under src/test because the plugin will
+    // fail if no schemas are found at the expected location
+    // ++ inConfig(Test)(baseGraphQLSettings)
 }
 
 object Codegen {
   def apply(
       outputDir: File,
       graphqlSchemaPaths: Seq[String],
```

Comment:
The only required attribute should be `outputDir`. All other attributes should be optional.

Reply:
Thanks for taking a look!

`graphqlSchemaPaths` is optional already since you can pass an empty list, which I think would be the best way to handle that one

For `schemasRootDir`, I just changed the methods using it to take an `Option[File]` so that I remember that it's optional. There are still some options from the Maven and Gradle plugins that haven't been implemented. `schemasRootDir` is one of them. I haven't added a check for it yet because it's currently always provided until I get around to implementing support for the rest of the options